packages/headless/src/features/product-listing/product-listing-actions.ts

```diff
@@ -92,12 +92,9 @@ export const fetchProductListing = createAsyncThunk<
   AsyncThunkProductListingOptions<StateNeededByFetchProductListing>
 >(
   'productlisting/fetch',
-  async (_action, {getState, dispatch, rejectWithValue, extra}) => {
-    const state = getState();
+  async (_action, {dispatch, rejectWithValue, extra}) => {
     const {apiClient} = extra;
-    const fetched = await apiClient.getProducts(
-      await buildProductListingRequest(state)
-    );
+    const fetched = await apiClient.getGraphQlProducts();
```

Comment:
I had to remove the state for this exemple as it is tied to the commerce-service context.